apps/dashboard/src/components/subscribers/subscriber-overview-form.tsx

```diff
@@ -0,0 +1,298 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { loadLanguage } from '@uiw/codemirror-extensions-langs';
+import { useForm } from 'react-hook-form';
+import { RiDeleteBin2Line } from 'react-icons/ri';
+import { z } from 'zod';
+import { Button } from '../primitives/button';
+import { Editor } from '../primitives/editor';
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from '../primitives/form/form';
+import { Input, InputRoot } from '../primitives/input';
+import { PhoneInput } from '../primitives/phone-input';
+import { Separator } from '../primitives/separator';
+import { SubscriberFormSchema } from './schema';
+import { TimezoneSelect } from './timezone-select';
+import { Avatar, AvatarFallback, AvatarImage } from '../primitives/avatar';
+import { Tooltip, TooltipContent, TooltipTrigger } from '../primitives/tooltip';
+import { Link } from 'react-router-dom';
+import { usePatchSubscriber } from '@/hooks/use-patch-subscriber';
+import { showSuccessToast } from '../primitives/sonner-helpers';
+import { SubscriberResponseDto } from '@novu/api/models/components';
+import { CopyButton } from '../primitives/copy-button';
+import { SubscriberOverviewSkeleton } from './subscriber-overview-skeleton';
+
+const extensions = [loadLanguage('json')?.extension ?? []];
+const basicSetup = { lineNumbers: true, defaultKeymap: true };
+
+export default function SubscriberOverviewForm({
+  subscriberId,
+  subscriber,
+  isFetching,
+}: {
+  subscriberId: string;
+  subscriber?: SubscriberResponseDto;
+  isFetching: boolean;
+}) {
+  const { patchSubscriber } = usePatchSubscriber({
+    onSuccess: () => {
+      showSuccessToast('Subscriber updated successfully');
+    },
+  });
+
+  /**
+   * Needed to forcefully reset the form when switching subscriber
+   * Without this, the form will keep the previous subscriber's data for undefined fields of current one
+   */
+  const subscriberDetails = isFetching ? undefined : subscriber;
+
+  const form = useForm<z.infer<typeof SubscriberFormSchema>>({
+    values: { ...subscriberDetails, data: JSON.stringify(subscriberDetails?.data, null, 2) },
+    resolver: zodResolver(SubscriberFormSchema),
+    shouldFocusError: false,
+  });
+
+  if (isFetching) {
+    return <SubscriberOverviewSkeleton />;
+  }
+
+  const onSubmit = async (formData: z.infer<typeof SubscriberFormSchema>) => {
+    const dirtyFields = form.formState.dirtyFields;
+
+    const dirtyPayload = Object.keys(dirtyFields).reduce<Partial<typeof formData>>((acc, key) => {
+      const typedKey = key as keyof typeof formData;
+      if (typedKey === 'data') {
+        return { ...acc, data: JSON.parse(formData.data) };
+      }
+      return { ...acc, [typedKey]: formData[typedKey] };
+    }, {});
+
+    if (!Object.keys(dirtyPayload).length) {
+      return;
+    }
+
+    await patchSubscriber({ subscriberId, subscriber: dirtyPayload });
+  };
+
+  return (
+    <div className="flex h-full flex-col items-stretch">
```

Comment:
Is this <div> necessary? Can we apply item-stretch to the <Form/> and have the same result?